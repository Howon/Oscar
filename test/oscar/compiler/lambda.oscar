def apply(f: [(int) => int], input: int) => int = {
  return f(input);
}

def main() => unit = {
  Println(apply((x:int) => int = x - 10, 11));

  int t = 5;

  def test() => unit = {
    def testest() => unit = {
      int b = 10;
      Println(b);
      Println(t);
    }
    testest();
    int b = 12;
    Println(b);
  }

  test();
}
