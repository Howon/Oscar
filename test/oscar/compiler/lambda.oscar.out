1
10
5
12
